compiler/codegen/symbol.cpp

```diff
@@ -2314,8 +2314,20 @@ static void pythonRetExternalArray(FnSymbol* fn, std::string& funcCall,
 
   funcCall += "cdef chpl_external_array ret_arr = ";
 
-  std::string typeStr = getPythonTypeName(eltType, PYTHON_PYX);
-  std::string typeStrCDefs = getPythonTypeName(eltType, C_PYX);
+  std::string typeStr;
+  std::string typeStrCDefs;
+
+  //
+  // String and bytes are not considered primitive types so we cannot fetch
+  // them from the Python type map.
+  //
+  if (eltType == dtString || eltType == dtBytes) {
```

Comment:
If dtString and dtBytes have been cleaned up by this point, why would we be able to check against them in this code?  I'm not necessarily sure it's wrong, but it struck me as odd

Reply:
I know, it is kind of odd...

But at the same time if it evaluates to true then they're guaranteed to be alive.

Reply:
Yeah, but we'd still be doing a comparison against something that is dead.  Have you done a valgrind run for the compiler in the cases where we don't expect matches?

Reply:
Yeah I did a bunch (10) on `interop/noLibFlag/pythonCompile`. Do you have a better test in mind? I could also run on `interop/python/multilocale/testing`.

Reply:
That seems fine.  Those were the ones that were causing problems before, right?

Reply:
To placate my own worry I'm going to try and redo the code to something that checks for if the types are alive, first.